### DISCUSSION TITLE
Adding a second TAP interface to SO

### DISCUSSION THREAD

**Discussion by rsharris3** [Security-Onion-Solutions/securityonion#3527] [Unsupported Versions]
Setup SO 2.3.30 with this CentOS7 ISO in a distributed environment on ESXi.  Have a manager, search and a forward sensor node.  We have 2 taps that we are looking to monitor with the forward sensor node.  During the install, both interfaces connected to the taps were selected as the monitor interfaces, which adds them to a round robin bond interface.  

We are seeing similar issues as https://github.com/Security-Onion-Solutions/securityonion/discussions/2676, when a second interface is added to bond0 we don't see traffic using tcpdump on the new interface when both interfaces are added to bond0.

When only one interface is up as part of the bond, tcpdump shows traffic on the bond0 and corresponding ens224 interface.  Interface ens256 only shows traffic with tcpdump when it is removed or down in the bond configuration.
```
$ sudo ifconfig
[sudo] password for 
bond0: flags=5443<UP,BROADCAST,RUNNING,PROMISC,MASTER,MULTICAST>  mtu 1500
        ether xx:xx:xx:xx:xx:xx  txqueuelen 1000  (Ethernet)
        RX packets 129049798  bytes 144290444649 (134.3 GiB)
        RX errors 0  dropped 12740  overruns 0  frame 0
        TX packets 0  bytes 0 (0.0 B)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

docker0: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet #.#.#.#  netmask 255.255.255.0  broadcast #.#.#.#
        ether dd:dd:dd:dd:dd:dd  txqueuelen 0  (Ethernet)
        RX packets 1577856  bytes 737119870 (702.9 MiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 1551088  bytes 290479154 (277.0 MiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

ens192: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet #.#.#.#  netmask 255.255.255.0  broadcast #.#.#.#
        ether xx:xx:xx:xx:xx:mm  txqueuelen 1000  (Ethernet)
        RX packets 559350  bytes 227578850 (217.0 MiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 546749  bytes 772295694 (736.5 MiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

ens224: flags=2499<UP,BROADCAST,RUNNING,NOARP,PROMISC,SLAVE>  mtu 1500
        ether xx:xx:xx:xx:xx:xx  txqueuelen 1000  (Ethernet)
        RX packets 2345381  bytes 1045624219 (997.1 MiB)
        RX errors 0  dropped 12892  overruns 0  frame 0
        TX packets 0  bytes 0 (0.0 B)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

ens256: flags=451<UP,BROADCAST,RUNNING,NOARP,PROMISC>  mtu 1500
        ether xx:xx:xx:xx:xx:YY  txqueuelen 1000  (Ethernet)
        RX packets 8514  bytes 8161317 (7.7 MiB)
        RX errors 0  dropped 70  overruns 0  frame 0
        TX packets 0  bytes 0 (0.0 B)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

```
When we have one interface as part of the bond, SO works as expected.    I noticed that when the bond is setup, the mode is Round-Robin and the bond mac address is the same as the primary monitor interface.  
![image](https://user-images.githubusercontent.com/80697363/111631262-9309c780-87c9-11eb-9eb3-64c1ed2ec9cf.png)
Is there a bond mode that will allow traffic from both interfaces to be monitored?  It doesn't appear to allow me to change the bond mode from the nmtui.
```
$ sudo nmcli con sho
[sudo] password for 
NAME               UUID                                  TYPE      DEVICE
ens192             3287ceb2-21fc-4f87-9e4f-149014f4f8b9  ethernet  ens192
bond0              f2881e16-9258-4636-b67b-9827791db018  bond      bond0
bond-slave-ens224  988d667c-afb2-46e0-8a92-96beb2124921  ethernet  ens224
bond-slave-ens256  53c007e0-e731-40a2-8bca-351926d7a701  ethernet  --
ens224             e15ca899-4010-4d76-8aad-fca50e91833c  ethernet  --
ens256             418c30d0-4d21-4181-9cd1-541d94263ed4  ethernet  --
```
Is a second bond interface needed and if so, how to we get SO to monitor that as well?  

Any help / suggestions is greatly appreciated.

**Comment by weslambert** [comment]
Security Onion setup configures a logical `bond0` interface for all chosen monitoring interfaces.  

In 2.3.30, there is a script called `so-monitor-add` which can be used like `so-monitor-add $interface` to add an additional interface to he the `bond0` configuration.

**Reply by rsharris3** [reply]
I get that, but when the second interface is added to `bond0` and brought up we don't see any traffic on the second tap interface.  If we issue `sudo nmcli con down 'bond-slave-ens256'` then use tcpdump on that interface, we see traffic again.  

I am not having an issue getting it configured as documented, it just doesn't work as expected when it is configured.

I was able to get it working, in the /etc/sysconfig/network-scripts/ifcfg-bond0 file change the bond mode from the default round-robin (balance-rr) to Adaptive load balancing (balance-alb).

```
BONDING_OPTS=mode=balance-alb
TYPE=Bond
BONDING_MASTER=yes
HWADDR=
MTU=1500
PROXY_METHOD=none
BROWSER_ONLY=no
IPV6INIT=no
NAME=bond0
UUID=
DEVICE=bond0
ONBOOT=yes
```
